src/lex.js

```diff
@@ -341,6 +341,14 @@ Lexer.prototype = {
       };
     }
 
+    // Exponentiation operator
+    if (ch1 === "*" && ch2 === "*") {
+      return {
+        type: Token.Punctuator,
+        value: "**"
+      };
+    }
+
     // 2-character punctuators: <= >= == != ++ -- << >> && ||
     // += -= *= %= &= |= ^= /=
     if (ch1 === ch2 && ("+-<>&|".indexOf(ch1) >= 0)) {
```

Comment:
Totally! I missed this opportunity because the surrounding comments are a
little inaccurate. I've updated those as well.